Unzip ais

MorganMasters opened the pull request.

MorganMasters committed:
build downloads._unzip_ais()

MorganMasters committed:
superficial docstring edits

Jacob-Stevens-Haas commented:
Confused by why you merged `download-ais` into this branch.  As I understand it, this PR is really just 32043e4 and 3189244.

Try checking out this branch and running `git reset 3189244` to undo the merge.  Then you will have to force push, `git push -f`.

Jacob-Stevens-Haas commented:
Ok good, that reset helps.  Now, I know we're talking about wget and other stuff on another branch, and the first commit here isn't really germane to this unzipping function.  So we can use the interactive rebase to remove that commit.

From this branch, try 'git rebase -i HEAD~4'.

You will get a text editor with the list of commits preceded by "pick".  You can then remove the spurious commit from this branch by changing "pick" to "drop" in front.  When you save and quit the editor, git rewinds the branch, then reapplies the diffs for each commit in the appropriate order.  

It's a powerful tool to reorder commits, combine "Work in progress" commits, and remover certain commits from branches. 💪

Jacob-Stevens-Haas committed:
CLN: Remove comment

Jacob-Stevens-Haas commented:
Ok things looked weird, so I fixed them.  For some reason you had two commits labeled "commit post reset", but idk what they were supposed to accomplish.  So I removed them.

Jacob-Stevens-Haas commented on tehom/downloads.py:
```diff
@@ -94,8 +95,9 @@ def _unzip_ais(zipfile: Path) -> Tuple[Path]:
-    pass
+    with ZipFile(f"{zipfile}", "r") as zipObj:
+        zipObj.extractall(_persistence.AIS_TEMP_DIR)
+    return _persistence.AIS_TEMP_DIR, zipfile.parts[-1]
```
First return value: We want to return the root of the unzip tree, not the folder that it's in.

Second return value: `zipfile` is a path to the zipped file.  We want to return the unzipped file's path.

MorganMasters replied:
how is this?

with ZipFile(f"{zipfile}", "r") as zipObj:
      zipObj.extractall(_persistence.AIS_TEMP_DIR)
  return _persistence.STORAGE, _persistence.AIS_TEMP_DIR / zipfile.parts[-1]

MorganMasters replied:
^ ignore that

Jacob-Stevens-Haas reviewed (changes requested):
A slight difference with return values.  Also, can you write a test for this?

MorganMasters commented:
By the root of the unzipped tree, do you mean the directory containing the
unzip file's directory?


On Thu, Aug 12, 2021 at 5:45 PM Jacob Stevens-Haas ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> A slight difference with return values. Also, can you write a test for
> this?
> ------------------------------
>
> In tehom/downloads.py
> <https://github.com/Jacob-Stevens-Haas/tehom/pull/7#discussion_r688171383>
> :
>
> > @@ -94,8 +95,9 @@ def _unzip_ais(zipfile: Path) -> Tuple[Path]:
>          tuple comprising the root of the unzip tree and the specific
>          unzipped file of interest
>      """
> -    # morgan
> -    pass
> +    with ZipFile(f"{zipfile}", "r") as zipObj:
> +        zipObj.extractall(_persistence.AIS_TEMP_DIR)
> +    return _persistence.AIS_TEMP_DIR, zipfile.parts[-1]
>
> First return value: We want to return the root of the unzip tree, not the
> folder that it's in.
>
> Second return value: zipfile is a path to the zipped file. We want to
> return the unzipped file's path.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Jacob-Stevens-Haas/tehom/pull/7#pullrequestreview-729179048>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMBKKIVWXDO4C5UJIJRSQ7TT4RTMBANCNFSM5B2QC5FA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

MorganMasters commented:
Changed to:

+    with ZipFile(f"{zipfile}", "r") as zipObj:
+        zipObj.extractall(_persistence.AIS_TEMP_DIR)
+    return _persistence.STORAGE, _persistence.AIS_TEMP_DIR / zipfile.parts[-1]

Is this what you mean?


On Sat, Aug 14, 2021 at 2:19 PM Morgan Masters ***@***.***>
wrote:

> By the root of the unzipped tree, do you mean the directory containing the
> unzip file's directory?
>
>
> On Thu, Aug 12, 2021 at 5:45 PM Jacob Stevens-Haas <
> ***@***.***> wrote:
>
>> ***@***.**** requested changes on this pull request.
>>
>> A slight difference with return values. Also, can you write a test for
>> this?
>> ------------------------------
>>
>> In tehom/downloads.py
>> <https://github.com/Jacob-Stevens-Haas/tehom/pull/7#discussion_r688171383>
>> :
>>
>> > @@ -94,8 +95,9 @@ def _unzip_ais(zipfile: Path) -> Tuple[Path]:
>>          tuple comprising the root of the unzip tree and the specific
>>          unzipped file of interest
>>      """
>> -    # morgan
>> -    pass
>> +    with ZipFile(f"{zipfile}", "r") as zipObj:
>> +        zipObj.extractall(_persistence.AIS_TEMP_DIR)
>> +    return _persistence.AIS_TEMP_DIR, zipfile.parts[-1]
>>
>> First return value: We want to return the root of the unzip tree, not the
>> folder that it's in.
>>
>> Second return value: zipfile is a path to the zipped file. We want to
>> return the unzipped file's path.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Jacob-Stevens-Haas/tehom/pull/7#pullrequestreview-729179048>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AMBKKIVWXDO4C5UJIJRSQ7TT4RTMBANCNFSM5B2QC5FA>
>> .
>> Triage notifications on the go with GitHub Mobile for iOS
>> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
>> or Android
>> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
>> .
>>
>

MorganMasters commented:
ah, wait, I see an issue

fixing




On Sat, Aug 14, 2021 at 2:28 PM Morgan Masters ***@***.***>
wrote:

> Changed to:
>
> +    with ZipFile(f"{zipfile}", "r") as zipObj:
> +        zipObj.extractall(_persistence.AIS_TEMP_DIR)
> +    return _persistence.STORAGE, _persistence.AIS_TEMP_DIR / zipfile.parts[-1]
>
> Is this what you mean?
>
>
> On Sat, Aug 14, 2021 at 2:19 PM Morgan Masters ***@***.***>
> wrote:
>
>> By the root of the unzipped tree, do you mean the directory containing
>> the unzip file's directory?
>>
>>
>> On Thu, Aug 12, 2021 at 5:45 PM Jacob Stevens-Haas <
>> ***@***.***> wrote:
>>
>>> ***@***.**** requested changes on this pull request.
>>>
>>> A slight difference with return values. Also, can you write a test for
>>> this?
>>> ------------------------------
>>>
>>> In tehom/downloads.py
>>> <https://github.com/Jacob-Stevens-Haas/tehom/pull/7#discussion_r688171383>
>>> :
>>>
>>> > @@ -94,8 +95,9 @@ def _unzip_ais(zipfile: Path) -> Tuple[Path]:
>>>          tuple comprising the root of the unzip tree and the specific
>>>          unzipped file of interest
>>>      """
>>> -    # morgan
>>> -    pass
>>> +    with ZipFile(f"{zipfile}", "r") as zipObj:
>>> +        zipObj.extractall(_persistence.AIS_TEMP_DIR)
>>> +    return _persistence.AIS_TEMP_DIR, zipfile.parts[-1]
>>>
>>> First return value: We want to return the root of the unzip tree, not
>>> the folder that it's in.
>>>
>>> Second return value: zipfile is a path to the zipped file. We want to
>>> return the unzipped file's path.
>>>
>>> —
>>> You are receiving this because you authored the thread.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/Jacob-Stevens-Haas/tehom/pull/7#pullrequestreview-729179048>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/AMBKKIVWXDO4C5UJIJRSQ7TT4RTMBANCNFSM5B2QC5FA>
>>> .
>>> Triage notifications on the go with GitHub Mobile for iOS
>>> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
>>> or Android
>>> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
>>> .
>>>
>>

Jacob-Stevens-Haas commented:
First off, your `git pull` merged the remote branch (github branch) with your local branch, reintroducing all the commits I removed.  On your local branch, run `git reset --hard 813cafb`.  Then `git push -f`

I also may have started some confusion; we need to stay using 2015 data and later, since data wasn't kept in csv format before 2015.  Using the example of January 2016, zone 1:
Downloaded filename: `zipfile=PosixPath('/home/xenophon/github/tehom/tehom/storage/ais/AIS_2016_01_Zone01.zip')`
When unzipped in place, you now have a file `../ais/AIS_ASCII_by_UTM_Month/2016/AIS_2016_01_Zone01.csv`
I would like the return values to be:
```
PosixPath('/home/xenophon/github/tehom/tehom/storage/ais/AIS_ASCII_by_UTM_Month'), PosixPath('/home/xenophon/github/tehom/tehom/storage/ais/AIS_ASCII_by_UTM_Month/2016/AIS_2016_01_Zone01.csv)
```
The reason is because the calling script needs to a) know the path to the csv in order to load it and (b), if successful, delete the unzipped directory.  Currently, the return values are 
```
PosixPath('/'), PosixPath('/home/xenophon/github/tehom/tehom/storage/ais/AIS_ASCII_by_UTM_Month/2016/AIS_2016_01_Zone01.csv')
```

Jacob-Stevens-Haas committed:
Merge branch 'master' into unzip-ais

Jacob-Stevens-Haas committed:
BUG: Fix typo in test data context manager

Jacob-Stevens-Haas committed:
ENH: Keep original filename when downloading AIS data

_unzip_ais needs to know the name of the file it unzips to, which is largely
the same filename but with csv.

Jacob-Stevens-Haas committed:
ENH: Add _unzip_ais() to downloads.py
